README.md

```diff
@@ -1,15 +1,89 @@
-- Simple API. Higher-level operations will be in helpers and fixtures, not the main `Juju` class.
-- No `async`. This was a "feature" of python-libjuju that isn't desired for integration tests.
+We designed Jubilant so it would:
+
+- Match the Juju CLI. Method, parameter, and response field names match the Juju CLI, with minor exceptions (such as "application" being shortened to "app").
+- Have a simple API. Higher-level operations will be in helpers and fixtures, not the main `Juju` class (the only exception being `Juju.wait`).
+- Not use `async`. This was a "feature" of python-libjuju that adds complexity and isn't needed for integration tests.
 - Support Juju 3 and 4. The Juju team is guaranteeing CLI arguments and `--format=json` responses won't change between Juju 3.x and 4.x. When Juju 5.x arrives and changes the CLI, we'll keep the Jubilant API simple and 1:1 with the 5.x CLI, but will consider adding a `jubilant.compat` layer to avoid tests have to manually work around differences between 4.x and 5.x.
+
+
+## Contributing and developing
+
+Anyone can contribute to Jubilant. It's best to start by [opening an issue](https://github.com/canonical/jubilant/issues) with a clear description of the problem or feature request, but you can also [open a pull request](https://github.com/canonical/jubilant/pulls) directly.
+
+Jubilant uses [`uv`](https://docs.astral.sh/uv/) to manage Python dependencies and tools, so you'll need to [install uv](https://docs.astral.sh/uv/#installation) to work on the library. You'll also need `make` to run local development tasks (but you probably have `make` installed already).
+
+After that, clone the Jubilant codebase and type `make all` to run various checks and the unit tests:
+
+```
+$ git clone https://github.com/canonical/jubilant
+Cloning into 'jubilant'...
+...
+$ cd jubilant
+$ make all
+...
+========== 107 passed in 0.26s ==========
+```
+
+To contribute a code change, write your fix or feature, add tests, document it, and run `make all` before you push and create a PR. Once you create a PR, GitHub will also run the integration tests, which takes several minutes.
+
+
+## Doing a release
+
+To create a new release of Jubilant:
+
+- Update the `__version__` field in [`jubilant/__init__.py`](https://github.com/canonical/jubilant/blob/main/jubilant/__init__.py) to the new version you want to release.
+- Push up a PR with this change and get it reviewed and merged.
+- Create a [new release](https://github.com/canonical/jubilant/releases/new) on GitHub with good release notes. The tag should start with a `v`, like `v1.2.3`. Once you've created the release, the [`publish.yaml` workflow](https://github.com/canonical/jubilant/blob/main/.github/workflows/publish.yaml) will automatically publish it to PyPI.
```

Comment:
Do you want to explicitly say that it's semver here? More for people consuming than people publishing.

Reply:
Good point. However, I'm going to add this to the little "Releases" section on the docs landing page (PR coming soon).